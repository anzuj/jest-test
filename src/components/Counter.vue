<template>
  <v-container>
    <p id="sample">cat</p>
    <h2>Prop count:</h2>
    <v-row align="center">
      <v-col cols="3">
        <div id="nr" class="mr-6">{{ count }}</div>
      </v-col>
      <v-col class="d-flex">
        <v-btn
          id="incrementBtn"
          class="mx-2"
          small
          outlined
          color="indigo"
          @click="increment($event)"
          ><v-icon>mdi-plus</v-icon> Increase prop
        </v-btn>
      </v-col>
    </v-row>
    <v-divider></v-divider>

    <h2>Local count:</h2>
    <v-row align="center">
      <v-col cols="3">
        <div id="nr" class="mr-6">{{ localcount }}</div>
      </v-col>
      <v-col class="d-flex">
        <v-btn
          id="incrementLocal"
          @click="localcount++"
          class="mx-2"
          small
          outlined
          color="indigo"
          ><v-icon>mdi-plus</v-icon> Increase local
        </v-btn>
      </v-col>
    </v-row>
  </v-container>
</template>

<script>
export default {
  props: ["count"],
  data: () => ({
    localcount: 0,
  }),
  methods: {
    increment() {
      this.$emit("increment");
    },
  },
};
</script>

<style>
#nr {
  font-size: 50px;
  font-weight: bold;
}
</style>
