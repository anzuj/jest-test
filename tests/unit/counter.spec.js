import Vue from "vue";
import Vuetify from "vuetify";

import Counter from "@/components/Counter.vue";

import { createLocalVue, mount } from "@vue/test-utils";

describe("Counter.vue", () => {
  const localVue = createLocalVue();
  let vuetify;

  beforeEach(() => {
    vuetify = new Vuetify();
  });

  // helper mount function to avoid repetition
  const mountFactory = (options) => {
    return mount(Counter, {
      localVue,
      vuetify,
      ...options,
    });
  };

  // TEST 1 ----------------------------------------------------
  it("Mounts properly", () => {
    const wrapper = mountFactory();
    expect(wrapper.findComponent(Counter).exists()).toBe(true);
  });

  // TEST 2 ----------------------------------------------------
  it("Count prop displays correctly", () => {
    const wrapper = mountFactory({
      propsData: { count: 5 },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  // TEST 3 isolated local increment-----------------------------
  it("Local increment", async() => {
    const wrapper = mountFactory();
    expect(wrapper.vm.localcount).toBe(0); //initial state

    wrapper.find("#incrementLocal").trigger("click");

    expect(wrapper.vm.localcount).toBe(1); //after click
  });

  // TEST 4 emits event to parent-----------------------------
  it("Emits increment event", () => {
    const wrapper = mountFactory();
    wrapper.vm.$emit("increment");

    // assert event has been emitted
    expect(wrapper.emitted().increment).toBeTruthy();
  });
});
