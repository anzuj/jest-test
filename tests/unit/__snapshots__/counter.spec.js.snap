// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter.vue Count prop displays correctly 1`] = `
<div class="container">
  <p id="sample">cat</p>
  <h2>Prop count:</h2>
  <div class="row align-center">
    <div class="col col-3">
      <div id="nr" class="mr-6">5</div>
    </div>
    <div class="d-flex col"><button type="button" class="mx-2 v-btn v-btn--depressed v-btn--flat v-btn--outlined theme--light v-size--small indigo--text" id="incrementBtn"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-plus theme--light"></i> Increase prop
      </span></button></div>
  </div>
  <hr role="separator" aria-orientation="horizontal" class="v-divider theme--light">
  <h2>Local count:</h2>
  <div class="row align-center">
    <div class="col col-3">
      <div id="nr" class="mr-6">0</div>
    </div>
    <div class="d-flex col"><button type="button" class="mx-2 v-btn v-btn--depressed v-btn--flat v-btn--outlined theme--light v-size--small indigo--text" id="incrementLocal"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-plus theme--light"></i> Increase local
      </span></button></div>
  </div>
</div>
`;
